wagtail/admin/wagtail_hooks.py

```diff
@@ -210,7 +210,9 @@ def page_listing_more_buttons(page, page_perms, is_parent=False, next_url=None):
         )
     if page_perms.can_delete():
         url = reverse('wagtailadmin_pages:delete', args=[page.id])
-        if next_url:
+
+        # After deleting the page, it is impossible to redirect to it.
+        if next_url and next_url != reverse('wagtailadmin_explore', args=[page.id]):
```

Comment:
Just an opinion, it feels like this bit of logic should be in its own block for added clarity to the comment:

```python
# After deleting the page, it is impossible to redirect to it.
if next_url != reverse('wagtailadmin_explore', args=[page.id]):
    next_url = None

if next_url:
```